sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -1663,15 +1665,28 @@ ProgramManager::link(const std::vector<device_image_plain> &DeviceImages,
   for (const device &Dev : Devs)
     PIDevices.push_back(getSyclObjImpl(Dev)->getHandleRef());
 
+  std::string LinkOptionsStr;
+  for (const device_image_plain &DeviceImage : DeviceImages) {
+    const std::shared_ptr<device_image_impl> &InputImpl =
+        getSyclObjImpl(DeviceImage);
+    const char *DeviceLinkOptions =
+        InputImpl->get_bin_image_ref()->getLinkOptions();
+    if (DeviceLinkOptions != nullptr) {
+      std::string TemporaryStr(DeviceLinkOptions);
+      std::optional<std::string> CompileOpts;
+      applyOptionsFromImage(*CompileOpts, TemporaryStr,
```

Comment:
It is [undefined behavior](https://en.cppreference.com/w/cpp/utility/optional/operator*) to use the `*` operator on an `std::optional` that has no value. Since `CompileOpts` is [created without a value](https://en.cppreference.com/w/cpp/utility/optional/optional), `*CompileOpts` here is undefined behavior.

Reply:
Could I set it to an empty string to define its behaviour? I mean change it to `std::optional<std::string> CompileOpts("");`

Reply:
You could, but I don't see a benefit to that over just having `std::string CompileOpts`.

Reply:
Yes, you are right.

Reply:
Changed to `std::string`